### PR TITLE
[apiRefactor] Remove all underscores in components/

### PR DESCRIPTION
**Pull request by rcchen**

### CODE REVIEW

File: src/components/axes/abstractAxis.ts [hunk 1]
```diff
@@ -14,14 +14,14 @@ module Plottable {
      * The css class applied to each tick label (the text associated with the tick).
      */
     public static TICK_LABEL_CLASS = "tick-label";
-    protected _tickMarkContainer: D3.Selection;
-    protected _tickLabelContainer: D3.Selection;
+    protected tickMarkContainer: D3.Selection;
+    protected tickLabelContainer: D3.Selection;
     protected _baseline: D3.Selection;
```
Comment (bluong) [inline src/components/axes/abstractAxis.ts]: I think this can remove the underscore

File: src/components/interpolatedColorLegend.ts [hunk 2]
```diff
@@ -3,19 +3,19 @@
...
   export class InterpolatedColorLegend extends Component {
-    private _measurer: SVGTypewriter.Measurers.Measurer;
-    private _wrapper: SVGTypewriter.Wrappers.Wrapper;
-    private _writer: SVGTypewriter.Writers.Writer;
-    private _scale: Scales.InterpolatedColor;
+    private measurer: SVGTypewriter.Measurers.Measurer;
+    private wrapper: SVGTypewriter.Wrappers.Wrapper;
+    private writer: SVGTypewriter.Writers.Writer;
+    private scale: Scales.InterpolatedColor;
     private _orientation: String ;
```
Comment (bluong) [inline src/components/interpolatedColorLegend.ts]: Can we make without underscore or save it for later if we rename the corresponding function to orientation?

File: src/components/plots/abstractPlot.ts [hunk 3]
```diff
@@ -232,8 +232,8 @@ module Plottable {
       return attrToAppliedProjector;
```
Comment (bluong) [inline src/components/plots/abstractPlot.ts]: What about line 202 in the file?

Comment (bluong) [inline src/components/plots/abstractPlot.ts]: Possible missing pieces here

File: src/components/plots/abstractStackedPlot.ts [hunk 4]
```diff
@@ -179,7 +179,7 @@ module Plottable {
       if (!primaryScale) {
         return;
       }
-      if (this._isAnchored && this._stackedExtent.length > 0) {
+      if (this.isAnchored && this._stackedExtent.length > 0) {
         primaryScale.updateExtent(this.getID().toString(), "_PLOTTABLE_PROTECTED_FIELD_STACK_EXTENT", this._stackedExtent);
       } else {
         primaryScale.removeExtent(this.getID().toString(), "_PLOTTABLE_PROTECTED_FIELD_STACK_EXTENT");
```
Comment (bluong) [inline src/components/plots/abstractStackedPlot.ts]: stacked has a bunch of missing pieces

**Comment by bluong** [comment]
Is it just the Component / Interactive component pieces?

**Comment by rcchen** [comment]
This is supposed to cover everything in src/components/ and is not fully through src/components/plots yet.

**Comment by bluong** [comment]
Due to our discussion, I think we should blow up this PR and adjust accordingly

**Comment by rcchen** [comment]
:bomb:

**Pull request closed without merging**